### PR TITLE
update (plugin/datadog) new metrics, tags and metrics flexible configurability

### PR DESCRIPTION
**Pull request by shashiranjan84**

### CODE REVIEW

File: app/plugins/datadog.md [hunk 1]
```diff
@@ -20,14 +20,13 @@ Log API metrics like request count, request size, response status and latency to
 
 ## Configuration
 
-Configuring the plugin is straightforward, you can add it on top of an [API][api-object] (or [Consumer][consumer-object]) by executing the following request on your Kong server:
+Configuring the plugin is straight forward, you can add it on top of an [API][api-object] (or [Consumer][consumer-object]) by executing the following request on your Kong server:
```
Comment (theetrain) [inline app/plugins/datadog.md]: 'straightforward' was correct.

File: app/plugins/datadog.md [hunk 2]
```diff
@@ -55,11 +54,39 @@ parameter                      | default     | description
...
+`status_count`             | tracks each status code returned as response | kong.\<api_name>.status.\<status>.count and kong.\<api_name>.status.\<status>.total
+`unique_users`             | tracks unique users made a request to the api | kong.\<api_name>.user.uniques
+`request_per_user`         | tracks request/user | kong.\<api_name>.user.\<consumer_id>.count
+`upstream_latency`         | tracks the time it took for the final service to process the request | kong.\<api_name>.upstream_latency
+`kong_latency`             | tracks the internal Kong latency that it took to run all the plugins | kong.\<api_name>.kong_latency
+`status_count_per_user`    | tracks request/status/user | kong.\<api_name>.user.\<customer_id>.status.\<status> and kong.\<api_name>.user.\<customer_id>.status.total
+
+### Metric fields
+
+Plugin can be configured with any combination of list of [Metrics](#metrics), each entry consist of following fileds.
```
Comment (theetrain) [inline app/plugins/datadog.md]: Sentence should read: "Plugin can be configured `with any combination of` [Metrics](#metrics), each entry `consists` of `the` following `fields`."

If the fields below are **required**, then the end of the sentence should read: "each entry `must contain` the following fields."

File: app/plugins/datadog.md [hunk 3]
```diff
@@ -55,11 +54,39 @@ parameter                      | default     | description
...
+`name`          | Datadog metric's name                                    | [Metrics](#metrics)          
+`stat_type`     | determines what sort of event the metric represents      | `gauge`, `timer`, `counter`, `histogram`, `meter` and `set`
+`sample_rate`<br>*conditional*   | sampling rate                           | `number`                 
+`customer_identifier`<br>*conditional*| authenticated user detail          | `consumer_id`, `custom_id`, `username`
+`tags`<br>*conditional*| List of tags                                      | `key[:value]`
+
+### Metric requirements
+
+1.  by default all metrics get logged.
+2.  metric with Stat type as `counter` or `gouge` must have `sample_rate` defined too.
```
Comment (theetrain) [inline app/plugins/datadog.md]: `too` -> `as well`

File: app/plugins/datadog.md [hunk 4]
```diff
@@ -55,11 +54,39 @@ parameter                      | default     | description
...
+3.  `unique_users` metric only works with Stat type `set`.
+4.  `status_count`, `status_count_per_user` and `request_per_user` work only with Stat type  as `counter`.
```
Comment (theetrain) [inline app/plugins/datadog.md]: Stat type -> `stat_type`

**Comment by shashiranjan84** [comment]
@theetrain thanks for reviewing it. I have made the advised changes, please review.

**Comment by theetrain** [comment]
@shashiranjan84 Please review my previous comment: https://github.com/Mashape/getkong.org/pull/312#discussion_r101377004
'field' was spelled incorrectly, and the end of that sentence needs clarification.

It may be clearer to the user if you change the text 'Stat type' to `stat_type` that way they can tell it's a variable, as mentioned in https://github.com/Mashape/getkong.org/pull/312#discussion_r101378282

**Comment by shashiranjan84** [comment]
Merge only after https://github.com/Mashape/kong/pull/2090 get merged.

**Pull request closed without merging**